src/utils/GroupCallUtils.ts

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { EventTimeline, MatrixClient, MatrixEvent, RoomState } from "matrix-js-sdk/src/matrix";
+import { deepCopy } from "matrix-js-sdk/src/utils";
+
+export const STUCK_DEVICE_TIMEOUT_MS = 1000 * 60 * 60; // 1 hour
+
+const CALL_STATE_EVENT_TYPE = "org.matrix.msc3401.call";
+const CALL_MEMBER_STATE_EVENT_TYPE = "org.matrix.msc3401.call.member";
+const CALL_STATE_EVENT_TERMINATED = "m.terminated";
+
+interface MDevice {
+    ["m.device_id"]: string;
+}
+
+interface MCall {
+    ["m.call_id"]: string;
+    ["m.devices"]: Array<MDevice>;
+}
+
+interface MCallMemberContent {
+    ["m.expires_ts"]: number;
+    ["m.calls"]: Array<MCall>;
+}
+
+const getRoomState = (client: MatrixClient, roomId: string): RoomState => {
+    return client.getRoom(roomId)
+        ?.getLiveTimeline()
+        ?.getState?.(EventTimeline.FORWARDS);
```

Comment:
```suggestion
    return client.getRoom(roomId)?.currentState;
```

Reply:
Yes, I know that. But then I found: https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/models/room.ts#L275-L277

Reply:
Why do we want go from the oldest?

Reply:
`FORWARDS` is the newest: https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/models/event-timeline.ts#L57-L61

Reply:
Seems you're right :) Those words really did confuse me...